From c96287bc1edad6846c66936664848eafe0cfc1d8 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Wed, 17 Dec 2025 20:00:21 +0800
Subject: [PATCH 025/243] CACHY: Use BFQ for SQ devices and mq-deadline for MQ
 devices

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 block/elevator.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index 71eeec5ab76f..f29f84ae00bd 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -744,8 +744,13 @@ void elevator_set_default(struct request_queue *q)
 	ctx.name = "adios";
 #else // !CONFIG_MQ_IOSCHED_DEFAULT_ADIOS
 	bool is_sq = q->nr_hw_queues == 1 || blk_mq_is_shared_tags(q->tag_set->flags);
+#if defined(CONFIG_CACHY) && defined(CONFIG_IOSCHED_BFQ)
+	if (is_sq)
+		ctx.name = "bfq";
+#else
 	if (!is_sq)
 		return;
+#endif /* CONFIG_CACHY && CONFIG_IOSCHED_BFQ */
 #endif // CONFIG_MQ_IOSCHED_DEFAULT_ADIOS
 
 	/*
-- 
2.55.0

