diff --git a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
--- a/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-oneplus-common.dtsi
@@ -237,8 +237,6 @@
 	wcn3990-pmu {
 		compatible = "qcom,wcn3990-pmu";
 
-		pinctrl-0 = <&sw_ctrl_default>;
-		pinctrl-names = "default";
 
 		vddio-supply = <&vreg_s4a_1p8>;
 		vddxo-supply = <&vreg_l7a_1p8>;
@@ -246,7 +244,6 @@
 		vddch0-supply = <&vreg_l25a_3p3>;
 		vddch1-supply = <&vreg_l23a_3p3>;
 
-		swctrl-gpios = <&pm8998_gpios 3 GPIO_ACTIVE_HIGH>;
 
 		regulators {
 			vreg_pmu_io: ldo0 {
@@ -887,13 +884,6 @@
 };
 
 &pm8998_gpios {
-	sw_ctrl_default: sw-ctrl-default-state {
-		pins = "gpio3";
-		function = "normal";
-
-		input-enable;
-		bias-pull-down;
-	};
 
 	volume_down_gpio: pm8998-gpio5-state {
 		pinconf {
