From 5048d7b6269e546b5dce596f2f2519d1994b9e9e Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Tue, 28 Apr 2026 08:45:32 +0200
Subject: [PATCH 119/211] drm/panel/visionox-rm69299: Split DSI commands out
 into enable/disable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

EDITME: Imported from f20260427-visionox-rm69299-enable-v1-0-fa7d62d26d87@sigxcpu.org
        Please review before sending.

The DSI commands should be sent in enable/disable, fix that which also
fixes various DSI commands failing on screen blank/unblank like:

  msm_dsi ae94000.dsi: [drm:dsi_cmds2buf_tx] *ERROR* wait for video done timed out
  dsi_cmds2buf_tx: cmd dma tx failed, type=0x5, data0=0x28, len=4, ret=-110
  panel-visionox-rm69299 ae94000.dsi.0: sending DCS SET_DISPLAY_OFF failed: -110

Even with this applied I'm seeing some

   disp_cc_mdss_pclk0_clk status stuck at 'on'
   disp_cc_mdss_byte0_intf_clk status stuck at 'on'
   disp_cc_mdss_byte0_clk status stuck at 'on'

on SHIFT6mq once during boot but all DSI timeouts are gone.

Signed-off-by: Guido Günther <agx@sigxcpu.org>

--- b4-submit-tracking ---
# This section is used internally by b4 prep for tracking purposes.
{
  "series": {
    "revision": 1,
    "change-id": "20260427-visionox-rm69299-enable-06f3d736f664",
    "prefixes": []
  }
}
-- 
2.55.0

