From 37e212369990f7cea7434976efd1f96f0c4ac2e3 Mon Sep 17 00:00:00 2001
From: Vasiliy Doylov <neko@altlinux.org>
Date: Tue, 1 Sep 2026 18:25:13 +0200
Subject: [PATCH 204/212] MAINTAINERS: Add entry for Onsemi LC89821X lens voice
 coil driver

Add entry for Onsemi LC89821X lens voice coil driver

Signed-off-by: Vasiliy Doylov <nekocwd@mainlining.org>
---
 MAINTAINERS | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index 2f6724901fc1..6c4305dce816 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -20426,6 +20426,13 @@ S:	Supported
 W:	http://www.onsemi.com
 F:	drivers/net/phy/ncn*
 
+ONSEMI LC89821X LENS VOICE COIL DRIVER
+M:	Vasiliy Doylov <nekocwd@mainlining.org>
+L:	linux-media@vger.kernel.org
+S:	Maintained
+F:	Documentation/devicetree/bindings/media/i2c/onnn,lc898217xc.yaml
+F:	drivers/media/i2c/lc89821x.c
+
 OP-TEE DRIVER
 M:	Jens Wiklander <jenswi@kernel.org>
 L:	op-tee@lists.trustedfirmware.org (moderated for non-subscribers)
-- 
2.55.0

