From 11a8f09e7f2313383851453285479f5c6843b39d Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Mon, 20 Jul 2026 12:06:59 +0200
Subject: [PATCH 13283/15333] MAINTAINERS: Add Matrix channel to the NFC
 subsystem

Community hang out there.

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index 8014b9f8253e..728228f1147e 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -19100,6 +19100,7 @@ F:	include/net/net_failover.h
 NFC SUBSYSTEM
 M:	David Heidelberg <david@ixit.cz>
 L:	oe-linux-nfc@lists.linux.dev
+C:	https://matrix.to/#/#linux-nfc:ixit.cz
 S:	Maintained
 T:	git https://codeberg.org/linux-nfc/linux.git
 F:	Documentation/devicetree/bindings/net/nfc/
-- 
2.55.0

