From e36dbad0882646443b347273111b50eeb66095ba Mon Sep 17 00:00:00 2001
From: Namjae Jeon <linkinjeon@kernel.org>
Date: Tue, 14 Jul 2026 18:48:40 +0900
Subject: [PATCH 14806/15333] MAINTAINERS: update mailing list address for ntfs

Add the newly created official mailing list for the ntfs.
This mailing list will be shared and used for both the kernel driver
and the ntfsprogs-plus utility project.

Signed-off-by: Namjae Jeon <linkinjeon@kernel.org>
---
 MAINTAINERS | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/MAINTAINERS b/MAINTAINERS
index 9f8d226b305a..4505498b01d9 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -19271,7 +19271,7 @@ F:	drivers/ntb/hw/intel/
 NTFS FILESYSTEM
 M:	Namjae Jeon <linkinjeon@kernel.org>
 M:	Hyunchul Lee <hyc.lee@gmail.com>
-L:	linux-fsdevel@vger.kernel.org
+L:	ntfs@lists.linux.dev
 S:	Maintained
 T:	git git://git.kernel.org/pub/scm/linux/kernel/git/linkinjeon/ntfs.git
 F:	Documentation/filesystems/ntfs.rst
-- 
2.55.0

